Tools/ardupilotwaf/chibios.py

```diff
@@ -624,17 +628,27 @@ def generate_hwdef_h(env):
         print(env.BOOTLOADER_OPTION)
         env.BOOTLOADER_OPTION += " --signed-fw"
         print(env.BOOTLOADER_OPTION)
-    hwdef_script = os.path.join(env.SRCROOT, 'libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py')
     hwdef_out = env.BUILDROOT
     if not os.path.exists(hwdef_out):
         os.mkdir(hwdef_out)
-    python = sys.executable
-    cmd = "{0} '{1}' -D '{2}' --params '{3}' '{4}'".format(python, hwdef_script, hwdef_out, env.DEFAULT_PARAMETERS, env.HWDEF)
+
+    # Build a list of the hardware definitions
+    hwdef_arr = [env.HWDEF]
     if env.HWDEF_EXTRA:
-        cmd += " '{0}'".format(env.HWDEF_EXTRA)
-    if env.BOOTLOADER_OPTION:
-        cmd += " " + env.BOOTLOADER_OPTION
-    return subprocess.call(cmd, shell=True)
+        # Add any extra hwdefs that may be defined
+        hwdef_arr.append(env.HWDEF_EXTRA)
```

Comment:
I think these should probably be `extend`, right?

Reply:
Same-same; one takes a list, the other  a single item. 

So yes - iff `env.HWDEF_EXTRA` is a list...

Reply:
I believe present behaviour in `chibios.py` is a single extra hwdef is provided to `chibios_hwdef.py`.

However, if I called `chibios_hwdef.py` directly, I could provide multiple hwdefs - they are received as a list and it is capable of setting multiple extra hwdefs.

That said, I'm not attempting to change behaviours here. This implementation is current behaviour.